#!/usr/bin/python
PACKAGE = 'recognition'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# NN-based recognition parameteres
gen.add('min_voting_faces',  int_t,    0, 'the minimum number of faces to be recognized', 1, 1, 20)
gen.add('min_support_faces', int_t,    0, 'the minimum number of faces which support the majority', 2, 1, 10)
gen.add('fp_threshold',      double_t, 0, 'the threshold for false positive detection', 0.75, 0.0, 2.0)
gen.add('num_neighbors',     int_t,    0, 'the number of neighbor points used for face classification', 5, 1, 20)

# Bayesian-based recognition parameters
gen.add('posterior_threshold',        double_t, 0, 'the minimum posterior to establish a face id', 0.3, 0.0, 1.0)
gen.add('num_observations_threshold', int_t,    0, 'the minimum number of observations to establish a face id', 2, 1, 20)
gen.add('neg_pdf_scale',              double_t, 0, 'the scaling parameter to model possibility that a negative pair makes a false positive observation', 1.0, 0.0, 10.0)

exit(gen.generate(PACKAGE, 'recognition', 'FaceRecognition'))
